### PR TITLE
Add separate forms for create and update Article

### PR DESCRIPTION
**Pull request by mmrshk**
- [x] Add CreateForm/UpdateForm
- [x] Add tests/factories
- [x] Add sidekiq
- [x] Add sidekiq job for batch article upload
- [x] Add batch article upload
- [x] Add broadcast with message on upload
- [x] Add destroy action 
- [x] Add patch update for Article

### CODE REVIEW

File: app/jobs/article_attachments_upload_job.rb [hunk 1]
```diff
@@ -0,0 +1,18 @@
+# frozen_string_literal: true
+
```
Comment (mmrshk) [inline app/jobs/article_attachments_upload_job.rb:2]: move to service?

File: app/forms/articles/base_form.rb [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+    private
+
+    def user_is_admin?
+      errors.add(:base, 'You are not authorised for this action') unless user.is_a?(Admin)
+    end
+
+    def tag_ids_exists?
+      @tags = []
+
+      errors.add(:base, 'Tag ids invalid') unless article_tags[:tag_ids].all? do |tag_id|
```
Comment (yuravenzhik) [inline app/forms/articles/base_form.rb]: refactor

File: app/forms/articles/create_form.rb [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+    end
+
+    private
+
+    def create_article_tags!
+      tags.each { |tag| article.article_tags.create!(tag: tag) }
+    end
+
+    def create_article!
+      @article = Article.create!(category: category, content: content, user: user)
```
Comment (yuravenzhik) [inline app/forms/articles/create_form.rb]: without @

File: app/controllers/articles_controller.rb [hunk 4]
```diff
@@ -21,13 +19,31 @@ def create
...
-  def destroy; end
+    if @form.save
+      redirect_to root_path, notice: 'Article is updated'
+    else
+      render :edit
+    end
+  end
+
+  def destroy
+    redirect_to root_path, notice: 'Article is deleted' if @article.destroy!
```
Comment (yuravenzhik) [inline app/controllers/articles_controller.rb]: handle else

File: app/jobs/article_attachments_upload_job.rb [hunk 5]
```diff
@@ -0,0 +1,18 @@
...
+class ArticleAttachmentsUploadJob < ApplicationJob
+  queue_as :default
+
+  def perform(attachment, current_user_id)
+    ActiveRecord::Base.transaction do
+      article = Article.create!(user_id: current_user_id)
```
Comment (yuravenzhik) [inline app/jobs/article_attachments_upload_job.rb]: create service object for controller attachment upload action and move db record creation and ArticleAttachmentsUploadJob creation there

File: app/views/articles/edit.html.erb [hunk 6]
```diff
@@ -0,0 +1,38 @@
+<p class="font-weight-bold"> Edit article </p>
```
Comment (yuravenzhik) [inline app/views/articles/edit.html.erb:1]: partial

File: app/controllers/article_attachments_controller.rb [hunk 7]
```diff
@@ -0,0 +1,13 @@
+# frozen_string_literal: true
+
+class ArticleAttachmentsController < ApplicationController
```
Comment (yuravenzhik) [inline app/controllers/article_attachments_controller.rb]: change

File: app/jobs/article_attachments_upload_job.rb [hunk 8]
```diff
@@ -5,12 +5,13 @@ class ArticleAttachmentsUploadJob < ApplicationJob
...
-      article.inactivate!
-      article_upload.destroy!
+      article = Article.inactive.build(user_id: current_user_id, content: article_upload.attachment.read)
+      article.upload_errors = ArticleValidator.new.validate(article)
+      article.save!
 
+      article_upload.destroy!
       ActionCable.server.broadcast('notice_channel', { id: article.id, content: 'article already procceeded' })
-    rescue ActiveRecord::RecordInvalid => e
+    rescue StandardError => e
```
Comment (yuravenzhik) [inline app/jobs/article_attachments_upload_job.rb]: -

File: app/jobs/article_attachments_upload_job.rb [hunk 9]
```diff
@@ -5,12 +5,13 @@ class ArticleAttachmentsUploadJob < ApplicationJob
 
   def perform(article_upload, current_user_id)
     ActiveRecord::Base.transaction do
-      article = Article.create!(user_id: current_user_id, content: article_upload.attachment.read)
-      article.inactivate!
-      article_upload.destroy!
+      article = Article.inactive.build(user_id: current_user_id, content: article_upload.attachment.read)
+      article.upload_errors = ArticleValidator.new.validate(article)
```
Comment (yuravenzhik) [inline app/jobs/article_attachments_upload_job.rb]: -

**Pull request merged**